plugins/modules/files/ini_file.py

```diff
@@ -367,9 +367,10 @@ def do_ini(module, filename, section=None, option=None, values=None,
                     section_lines = new_section_lines
         else:
             # drop the entire section
-            section_lines = []
-            msg = 'section removed'
-            changed = True
+            if len(section_lines) != 0:
```

Comment:
More Pythonic would be:
```suggestion
            if section_lines:
```